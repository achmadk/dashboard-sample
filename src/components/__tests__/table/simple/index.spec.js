import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import chai from 'chai'

import {
  loadTableContent,
  SimpleTable,
  styles
} from '@/components/table/simple'

describe('Simple Table Component', () => {
  it('render correctly with loading props value is true', () => {
    const wrapper = shallow(<SimpleTable classes={styles()} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('render correctly with loading props value is false', () => {
    const wrapper = shallow(<SimpleTable classes={styles()} loading={false} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('return Table Content component successfully', () => {
    const result = loadTableContent()
    chai.expect(result).to.exist
  })

  it('return style successfully', () => {
    const result = styles()
    chai.expect(result).to.exist
  })
})
