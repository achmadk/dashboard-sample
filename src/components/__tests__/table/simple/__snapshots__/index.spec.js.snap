// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Table Component render correctly with loading props value is false 1`] = `
<WithStyles(Paper)
  className={
    Object {
      "marginBottom": 16,
      "overflowX": "auto",
    }
  }
  style={
    Object {
      "height": "100%",
    }
  }
>
  <React.Suspense
    fallback={
      <Skeleton
        animated={true}
        borderRadius="8px"
        color="#EFF1F6"
        count={25}
        height={150}
        heightRandomness={0}
        width="100%"
        widthRandomness={0.25}
      />
    }
  >
    <React.Lazy
      classes={
        Object {
          "tableRoot": Object {
            "marginBottom": 16,
            "overflowX": "auto",
          },
          "title": Object {
            "flex": "0 0 auto",
          },
        }
      }
      initialHeight={450}
      loading={false}
      title="Simple Table Title"
    />
  </React.Suspense>
</WithStyles(Paper)>
`;

exports[`Simple Table Component render correctly with loading props value is true 1`] = `
<WithStyles(Paper)
  className={
    Object {
      "marginBottom": 16,
      "overflowX": "auto",
    }
  }
  style={
    Object {
      "height": 450,
    }
  }
>
  <React.Suspense
    fallback={
      <Skeleton
        animated={true}
        borderRadius="8px"
        color="#EFF1F6"
        count={25}
        height={150}
        heightRandomness={0}
        width="100%"
        widthRandomness={0.25}
      />
    }
  >
    <React.Lazy
      classes={
        Object {
          "tableRoot": Object {
            "marginBottom": 16,
            "overflowX": "auto",
          },
          "title": Object {
            "flex": "0 0 auto",
          },
        }
      }
      initialHeight={450}
      loading={true}
      title="Simple Table Title"
    />
  </React.Suspense>
</WithStyles(Paper)>
`;
